client/prometheus.go

```diff
@@ -51,6 +51,31 @@ func (c *Prometheus) Close() error {
 	return nil
 }
 
+// buildIncrement calls incrementer factory if incrementer was not created before
+func (c *Prometheus) buildIncrement(metric, metricTotal string) {
+	c.Lock()
+	defer c.Unlock()
+
+	if _, ok := c.increments[metric]; !ok {
+		c.increments[metric] = c.incFactory.Create()
+	}
+
+	if _, ok := c.increments[metricTotal]; !ok {
+		c.increments[metricTotal] = c.incFactory.Create()
+	}
+
+}
+
+// buildState calls state factory if state objects was not created before
+func (c *Prometheus) buildState(metric string) {
```

Comment:
same as increment - work with the map only here, the rest should use reference to an object